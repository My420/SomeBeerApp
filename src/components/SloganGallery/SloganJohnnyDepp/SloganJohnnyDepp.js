import React from 'react';
import styles from './SloganJohnnyDepp.module.scss';
import Picture from '../../Picture/Picture';
import imagePC from './img/beer_beach@PC.jpg';
import imageTablet from './img/beer_beach@Tablet.jpg';
import imageMobile from './img/beer_beach@Mobile.jpg';

const SloganJohnnyDepp = () => {
  // eslint-disable-next-line no-console
  console.log('render ===== SloganJohnnyDepp');

  return (
    <article className={styles.container}>
      <h5 className="visually-hidden"> Johnny Depp quote</h5>
      <Picture
        className={styles.image}
        alt="beer cheer"
        pcSrc={imagePC}
        tabletSrc={imageTablet}
        mobileSrc={imageMobile}
      />
      <blockquote className={styles.quote}>
        <p className={styles.text}>
          «I&apos;m an old-fashioned guy... I want to be an old man with a beer
          belly sitting on a porch, looking at a lake or something»
        </p>
        <cite className={styles.author}>~ Johnny Depp</cite>
      </blockquote>
    </article>
  );
};

export default SloganJohnnyDepp;
