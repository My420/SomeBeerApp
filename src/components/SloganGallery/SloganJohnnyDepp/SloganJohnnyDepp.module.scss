@import '../../../sass/variables.scss';

.container {
  position: relative;
  width: 100%;
  height: 100%;
}

.image {
  width: 100%;
  height: 100%;
}

.quote {
  position: absolute;

  top: 0px;
  left: 0px;

  display: flex;
  flex-direction: column;
  align-items: flex-end;

  width: 100%;
}

.text,
.author {
  padding: 10px 20px 0 10px;

  font-size: 22px;
  line-height: 26px;
  font-weight: 700;
}

.author {
  align-self: flex-start;
}

@media (max-width: $MPC-max-width) {
  .text,
  .author {
    font-size: 20px;
    line-height: 24px;
  }
}

@media (max-width: $Tablet9-max-width) {
  .text,
  .author {
    padding: 5px 10px 0 10px;

    font-size: 18px;
    line-height: 22px;
  }
}

@media (max-width: 650px) {
  .text,
  .author {
    font-size: 16px;
    line-height: 20px;
  }
}

@media (max-width: 550px) {
  .text,
  .author {
    font-size: 14px;
    line-height: 18px;
  }
}

@media (max-width: $Tablet7-max-width) {
  .text,
  .author {
    font-size: 10px;
    line-height: 14px;
  }

  .author {
    padding: 0px 10px 0 5px;
  }
}

@media (max-width: 360px) {
  .text,
  .author {
    font-size: 10px;
    line-height: 14px;
  }
}
