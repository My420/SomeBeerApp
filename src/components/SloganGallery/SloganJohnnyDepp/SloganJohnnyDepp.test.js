import React from 'react';
import { shallow } from 'enzyme';
import SloganJohnnyDepp from './SloganJohnnyDepp';

describe('Test <SloganJohnnyDepp/> component', () => {
  const wrapper = shallow(<SloganJohnnyDepp />);

  test('should correct render component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
