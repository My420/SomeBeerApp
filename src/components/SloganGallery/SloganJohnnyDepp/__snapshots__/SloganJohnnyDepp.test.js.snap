// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <SloganJohnnyDepp/> component should correct render component 1`] = `
<article
  className="container"
>
  <h5
    className="visually-hidden"
  >
     Johnny Depp quote
  </h5>
  <Picture
    alt="beer cheer"
    className="image"
    mobileSrc="beer_beach@Mobile.jpg"
    pcSrc="beer_beach@PC.jpg"
    tabletSrc="beer_beach@Tablet.jpg"
  />
  <blockquote
    className="quote"
  >
    <p
      className="text"
    >
      «I'm an old-fashioned guy... I want to be an old man with a beer belly sitting on a porch, looking at a lake or something»
    </p>
    <cite
      className="author"
    >
      ~ Johnny Depp
    </cite>
  </blockquote>
</article>
`;
